components/threading/common/follow_button/follow_button.tsx

```diff
@@ -1,40 +1,37 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import React, {memo, ComponentProps, MouseEventHandler} from 'react';
-import {FormattedMessage} from 'react-intl';
+import React, {memo, ComponentProps} from 'react';
+import {useIntl} from 'react-intl';
+
+import classNames from 'classnames';
 
 import Button from '../button';
+import {t} from 'utils/i18n';
 
 type Props = {
-    isFollowing: boolean;
-    follow: MouseEventHandler<HTMLButtonElement>;
-    unFollow: MouseEventHandler<HTMLButtonElement>;
+    isFollowing: boolean | null | undefined;
```

Comment:
You're right that we don't technically need `undefined`, I'll go ahead and remove that. But when null (while thread data is loading), the follow button is disabled to prevent errant changes/toggles.